src/Components/Categories/CategoriesForm.js

```diff
@@ -1,205 +1,300 @@
+	const id = props?.data?.id
+	const [initialValues, setInitialValues] = useState({
+		name: categories?.name ? categories.name : '',
+		description: categories?.description ? categories.description : '',
+		image: React.createRef(),
+	})
 
-    const categories = props?.data
-    const categoriesBtn = props?.btnCat //si es true viene del componente crear, si es false de actualziar
-    const id = props?.data?.id
-    const [initialValues, setInitialValues] = useState({
-        name: categories?.name ? categories.name : '',
-        description: categories?.description ? categories.description : '',
-        image: React.createRef()
-    })
+	const [imgPost, setImgPost] = useState('')
 
-    const [imgPost,setImgPost] = useState('')
+	const bodyFormCategory = [
+		{
+			name: initialValues.name,
+			description: initialValues.description,
+			image: imgPost,
+		},
+		{
+			name: initialValues.name,
+			description: initialValues.description,
+		},
+	]
 
-    const bodyFormCategory = [    
-    {
-        name: initialValues.name, 
-        description: initialValues.description, 
-        image: imgPost,
-    },
-    {
-        name: initialValues.name, 
-        description: initialValues.description, 
-    }]
+	const [messageAlert, setMessageAlert] = useState({
+		msjFormIncomplete: false,
+		msjFormIncompleteImg: false,
+		msjFormIncompleteDes: false,
+		loandingForm: false,
+		messageErrorCategory: false,
+		messageOkCategory: false,
+	})
 
-    const [messageAlert, setMessageAlert] = useState({
-        msjFormIncomplete: false,
-        msjFormIncompleteImg: false,
-        msjFormIncompleteDes: false,
-        loandingForm: false,
-        messageErrorCategory: false,
-        messageOkCategory: false,
-    })
+	function getBase64(file) {
+		const reader = new FileReader()
+		reader.readAsDataURL(file)
+		reader.onload = function () {
+			setImgPost(reader.result)
+			setMessageAlert({ ...messageAlert, msjFormIncompleteImg: false })
+		}
+		reader.onerror = function (error) {
+			console.log('Error: ', error)
+		}
+	}
 
+	const handleChange = (e) => {
+		if (e.target.name === 'name') {
+			setInitialValues({ ...initialValues, name: e.target.value })
+			setMessageAlert({ ...messageAlert, msjFormIncomplete: false })
+		}
+	}
 
-    function getBase64(file) { 
-        var reader = new FileReader(); 
-        reader.readAsDataURL(file); 
-        reader.onload = function () { 
-            setImgPost(reader.result)
-            setMessageAlert({...messageAlert, msjFormIncompleteImg: false})
-        }; 
-        reader.onerror = function (error) {
-             console.log('Error: ', error);
-        }; 
-    }
+	const handleCkreditor = (e, editor) => {
+		const data = editor.getData()
+		setInitialValues({ ...initialValues, description: data })
+		setMessageAlert({ ...messageAlert, msjFormIncompleteDes: false })
+	}
 
+	const handleSubmit = (e) => {
+		e.preventDefault()
+	}
+	// eslint-disable-next-line
+	const handleClick = async () => {
```

Comment:
Esta función me estaba arrojando el error de: **Refactor this function to reduce its Cognitive Complexity from 47 to the 15 allowed.eslint [sonarjs/cognitive-complexity](https://github.com/SonarSource/eslint-plugin-sonarjs/blob/master/docs/rules/cognitive-complexity.md)**

Traté de ajustarla pero me fue bastante complejo entenderla, por lo que necesitaría ayuda del que la creó originalmente.